test/json-checker.jl

```diff
@@ -0,0 +1,35 @@
+# Run JSON checker tests, excluding tests fail01 and fail18
+
+const JSON_DATA_DIR = joinpath(Pkg.dir("JSON"), "data")
```

Comment:
don't use Pkg.dir, the package may be installed and used from somewhere else. `dirname(@__FILE__)` is better